### PR TITLE
fix memory leak in spine

### PR DESCRIPTION
**Pull request by rablwupei**

### CODE REVIEW

File: extensions/spine/SkeletonJson.cpp [hunk 1]
```diff
@@ -240,7 +240,7 @@ SkeletonData* SkeletonJson_readSkeletonDataFile (SkeletonJson* self, const char*
 		return 0;
 	}
 	skeletonData = SkeletonJson_readSkeletonData(self, json);
-	FREE(json);
+    delete [] json;
```
Comment (minggo) [inline extensions/spine/SkeletonJson.cpp:243]: But json is created by `NEW`, and the definition of `NEW` is 

```
   #define NEW(TYPE) CALLOC(TYPE,1)
```

So use `FREE` here is correct.

Comment (rablwupei) [inline extensions/spine/SkeletonJson.cpp:243]: json create in `const char* json = _Util_readFile(path, &length);`

in this function it use:

```
char* _Util_readFile (const char* path, int* length) {
    unsigned long size;
    char* data = reinterpret_cast<char*>(FileUtils::getInstance()->getFileData(
        FileUtils::getInstance()->fullPathForFilename(path).c_str(), "r", &size));
    *length = size;
    return data;
}
```

`getFileData()` need `delete []`

Comment (minggo) [inline extensions/spine/SkeletonJson.cpp:243]: Sorry, i thought `skeletonData`.
You are right, it is a memory leak.

**Comment by pkxpp** [comment]
the new version is still FREE, not delete, why not modify?

**Pull request merged**